From 18f185a4def90b59ee46f34e8264d2bb0885cce7 Mon Sep 17 00:00:00 2001
From: Fabian Kosmale <fabian.kosmale@qt.io>
Date: Tue, 26 Aug 2025 11:29:55 +0200
Subject: [PATCH 051/239] CRA review: mark qml/qmlmeta subfolder

Everything gets default significance. Note: there's a QML custom parser
in usage, but that's not "real" parsing, and we document that QML
documents have to be from a trusted source.

Pick-to: 6.9 6.8
Fixes: QTBUG-136202
Change-Id: Ifbc5a2a33cc527c14dc5aa328da8f634f63df5f8
Reviewed-by: Ulf Hermann <ulf.hermann@qt.io>
(cherry picked from commit 4a60da23f67f725b68d5947fc51b41ba1e08f87f)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/qmlmeta/types/qqmlbind.cpp             | 1 +
 src/qmlmeta/types/qqmlbind_p.h             | 1 +
 src/qmlmeta/types/qqmlconnections.cpp      | 1 +
 src/qmlmeta/types/qqmlconnections_p.h      | 1 +
 src/qmlmeta/types/qqmllocaleenums_p.h      | 1 +
 src/qmlmeta/types/qqmlloggingcategory.cpp  | 1 +
 src/qmlmeta/types/qqmlloggingcategory_p.h  | 1 +
 src/qmlmeta/types/qqmlmetadependencies.cpp | 1 +
 src/qmlmeta/types/qqmlmetadependencies_p.h | 1 +
 src/qmlmeta/types/qqmltimer.cpp            | 1 +
 src/qmlmeta/types/qqmltimer_p.h            | 1 +
 11 files changed, 11 insertions(+)

diff --git a/src/qmlmeta/types/qqmlbind.cpp b/src/qmlmeta/types/qqmlbind.cpp
index a66ace6ddd..579bc7e095 100644
--- a/src/qmlmeta/types/qqmlbind.cpp
+++ b/src/qmlmeta/types/qqmlbind.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2024 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #include "qqmlbind_p.h"
 
diff --git a/src/qmlmeta/types/qqmlbind_p.h b/src/qmlmeta/types/qqmlbind_p.h
index 970f7b143d..634691778d 100644
--- a/src/qmlmeta/types/qqmlbind_p.h
+++ b/src/qmlmeta/types/qqmlbind_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2016 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #ifndef QQMLBIND_H
 #define QQMLBIND_H
diff --git a/src/qmlmeta/types/qqmlconnections.cpp b/src/qmlmeta/types/qqmlconnections.cpp
index 8baefe0b8e..9754e8aba9 100644
--- a/src/qmlmeta/types/qqmlconnections.cpp
+++ b/src/qmlmeta/types/qqmlconnections.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2016 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #include "qqmlconnections_p.h"
 
diff --git a/src/qmlmeta/types/qqmlconnections_p.h b/src/qmlmeta/types/qqmlconnections_p.h
index 84bfe1171a..fd166a571e 100644
--- a/src/qmlmeta/types/qqmlconnections_p.h
+++ b/src/qmlmeta/types/qqmlconnections_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2016 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #ifndef QQMLCONNECTIONS_H
 #define QQMLCONNECTIONS_H
diff --git a/src/qmlmeta/types/qqmllocaleenums_p.h b/src/qmlmeta/types/qqmllocaleenums_p.h
index 771b74e5fb..7c96a09488 100644
--- a/src/qmlmeta/types/qqmllocaleenums_p.h
+++ b/src/qmlmeta/types/qqmllocaleenums_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2024 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #ifndef QQMLLOCALEENUMS_H
 #define QQMLLOCALEENUMS_H
diff --git a/src/qmlmeta/types/qqmlloggingcategory.cpp b/src/qmlmeta/types/qqmlloggingcategory.cpp
index 2f35203701..e3c3621fb6 100644
--- a/src/qmlmeta/types/qqmlloggingcategory.cpp
+++ b/src/qmlmeta/types/qqmlloggingcategory.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2016 Pelagicore AG
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #include "qqmlloggingcategory_p.h"
 
diff --git a/src/qmlmeta/types/qqmlloggingcategory_p.h b/src/qmlmeta/types/qqmlloggingcategory_p.h
index fa79383dd5..dd88645b9c 100644
--- a/src/qmlmeta/types/qqmlloggingcategory_p.h
+++ b/src/qmlmeta/types/qqmlloggingcategory_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2016 Pelagicore AG
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #ifndef QQMLLOGGINGCATEGORY_P_H
 #define QQMLLOGGINGCATEGORY_P_H
diff --git a/src/qmlmeta/types/qqmlmetadependencies.cpp b/src/qmlmeta/types/qqmlmetadependencies.cpp
index 1023f0f468..8f398ddc03 100644
--- a/src/qmlmeta/types/qqmlmetadependencies.cpp
+++ b/src/qmlmeta/types/qqmlmetadependencies.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2024 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #include "qqmlmetadependencies_p.h"
 #include <private/qtqmlglobal_p.h>
diff --git a/src/qmlmeta/types/qqmlmetadependencies_p.h b/src/qmlmeta/types/qqmlmetadependencies_p.h
index 2175cc2cd9..ee2e217a1c 100644
--- a/src/qmlmeta/types/qqmlmetadependencies_p.h
+++ b/src/qmlmeta/types/qqmlmetadependencies_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2024 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #ifndef QQMLMETADEPENDENCIES_P_H
 #define QQMLMETADEPENDENCIES_P_H
diff --git a/src/qmlmeta/types/qqmltimer.cpp b/src/qmlmeta/types/qqmltimer.cpp
index bb7e0da7fa..d0f16d1a6e 100644
--- a/src/qmlmeta/types/qqmltimer.cpp
+++ b/src/qmlmeta/types/qqmltimer.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2016 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #include "qqmltimer_p.h"
 
diff --git a/src/qmlmeta/types/qqmltimer_p.h b/src/qmlmeta/types/qqmltimer_p.h
index f926262952..050ac1eeda 100644
--- a/src/qmlmeta/types/qqmltimer_p.h
+++ b/src/qmlmeta/types/qqmltimer_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2016 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #ifndef QQMLTIMER_H
 #define QQMLTIMER_H
-- 
2.51.2

